### PR TITLE
Bug/134/is master

### PR DESCRIPTION
**Pull request by Jhalter5Stones**

### CODE REVIEW

**Comment by Jhalter5Stones** [comment]
@mrogers-5s Ready for review. Let me know what you think

https://trello.com/c/yolBI8jk/134-asbat-see-products-syncd-correctly-connecting-ismaster-to-custom-field-in-bc

File: lib/mapper/product_mapper.rb [hunk 1]
```diff
@@ -158,6 +160,19 @@ def self.get_variant_by_sku(sku, product)
...
+                    additionalProperty = clone['model'].select {|a| a['additionalProperty']}
+                    additionalProperty.each do |prop|
+                        if clone['model'].select {|a| a['isDefault']}
+                            prop.select {|a| a['additionalProperty']}['additionalProperty'].each do |a|
+                              clone['additionalProperty'].push(a)
+                            end
+                        end
+                    end
+                end
+            end
```
[the remark this answers is not in the document]
Comment (Jhalter5Stones) [reply lib/mapper/product_mapper.rb]: @mrogers-5s Fixed let me know what you think

https://trello.com/c/yolBI8jk/134-asbat-see-products-syncd-correctly-connecting-ismaster-to-custom-field-in-bc

File: lib/mapper/product_mapper.rb [hunk 2]
```diff
@@ -158,6 +160,24 @@ def self.get_variant_by_sku(sku, product)
...
+                                        if a['propertyID'] == field[key]
+                                            clone['additionalProperty'].push(a)
+                                        end
+                                    end
+                                end
+                            end
+                        end
+                    end
+                end
+            end
```
Comment (mrogers-5s) [inline lib/mapper/product_mapper.rb:180]: @Jhalter5Stones From an efficiency standpoint, I would probably _start_ with the values in `fieldMap`:

```
defaultVariant = clone['model'].select { |v| v['isDefault'] }
if defaultVariant && defaultVariant['additionalProperty']
    field_map['additionalProperty'].each do |field, key|
        prop = defaultVariant['additionalProperty'].select { |prop| prop['propertyID'] == field[key] }
        clone['additionalProperty'].push(prop) unless prop.nil?
    unless  field_map.nil? || field_map['additionalProperty'].nil?
end
```

**Comment by Jhalter5Stones** [comment]
@jspizziri Ready for final review. If you think this is good and get it merged can you also push to ruby gems? Thank you!

https://trello.com/c/yolBI8jk/134-asbat-see-products-syncd-correctly-connecting-ismaster-to-custom-field-in-bc

**Comment by jspizziri** [comment]
@Jhalter5Stones released as `v1.3.1`.

**Pull request merged**